### ISSUE TITLE
TypeError: not all arguments converted during string formatting

### ISSUE THREAD

**Issue by aroberge** [friendly-traceback/friendly-traceback#113]
See: https://twitter.com/reuvenmlerner/status/1486346697336889344

**Comment by alexmojaki** [comment]
Should be able to get the `%` node via executing and hopefully at least one operand with `pure_eval`. If you can get the left, and it's a string, and trying to `%` it yourself against a single value gives the same error, then you know that it contains no formatting placeholders like `%s` at all. Otherwise, if you can get the right, and it's not a tuple, same conclusion. Either way, you can then guess that they either made the same mistake as in the tweet if the string doesn't contain `%` at all, or that they messed up writing a `%` placeholder. On the other hand, if it seems that the format string has multiple placeholders, then they probably made a more 'normal' mistake and just have more placeholders than formatting arguments.

**Comment by aroberge** [comment]
Yes, I do get the node using `executing and  `pure_eval` is used to extract the information about `n` in
![image](https://user-images.githubusercontent.com/629698/183219471-d7b33736-81db-453b-9ff3-59a4ee7f2acc.png)

So, I definitely should be able to do what you describe. I just need to sit down and write various examples with corresponding error message for each case.
